test/verifier/verifiers.test.ts

```diff
@@ -0,0 +1,352 @@
+    rskProvider = new providers.JsonRpcProvider(RSK_URL);
+  });
+
+  describe('Deploy verifier', function () {
+    let deployVerifier: DeployVerifier;
+    let owner: Wallet;
+    let testToken: TestToken;
+    let relayHub: SignerWithAddress;
+    let smartWalletFactory: SmartWalletFactory;
+
+    beforeEach(async function () {
+      const [, localRelayHub] = await hardhat.getSigners();
+      relayHub = localRelayHub as SignerWithAddress;
+
+      owner = Wallet.createRandom().connect(rskProvider);
+
+      const hardHatSmartWalletFactory = await hardhat.getContractFactory(
+        'SmartWallet'
+      );
+      const smartWalletTemplate = await hardHatSmartWalletFactory.deploy();
+
+      const hardHatWalletFactory = await hardhat.getContractFactory(
+        'SmartWalletFactory'
+      );
+
+      smartWalletFactory = await hardHatWalletFactory.deploy(
+        smartWalletTemplate.address
+      );
+
+      testToken = (await prepareToken('TestToken')) as TestToken;
+
+      const expectedAddress = await smartWalletFactory.getSmartWalletAddress(
+        owner.address,
+        constants.AddressZero,
+        SMART_WALLET_INDEX
+      );
+
+      await testToken.mint(TOKEN_AMOUNT_TO_TRANSFER + 10, expectedAddress);
+
+      const deployVerifierFactory = await hardhat.getContractFactory(
+        'DeployVerifier'
+      );
+      deployVerifier = await deployVerifierFactory.deploy(
+        smartWalletFactory.address
+      );
+
+      await deployVerifier.acceptToken(testToken.address);
+    });
+
+    it('Should succeed when the deploy is correct', async function () {
+      const deployRequest = createEnvelopingRequest(
+        true,
+        {
+          relayHub: relayHub.address,
+          from: owner.address,
+          tokenAmount: TOKEN_AMOUNT_TO_TRANSFER,
+          tokenContract: testToken.address,
+          tokenGas: 50000,
+        },
+        {
+          callForwarder: smartWalletFactory.address,
+          callVerifier: deployVerifier.address,
+        }
+      ) as DeployRequest;
+
+      const signature = '0x00';
+
+      await expect(deployVerifier.verifyRelayedCall(deployRequest, signature))
+        .not.to.be.rejected;
+    });
+
+    it('Should fail if there is an  smartWallet already deployed at that address', async function () {
+      const deployRequest = createEnvelopingRequest(
+        true,
+        {
+          relayHub: relayHub.address,
+          from: owner.address,
+          tokenAmount: TOKEN_AMOUNT_TO_TRANSFER,
+          tokenContract: testToken.address,
+          tokenGas: 50000,
+        },
+        {
+          callForwarder: smartWalletFactory.address,
+        }
+      ) as DeployRequest;
+
+      const signature = '0x00';
+
+      await createSupportedSmartWallet({
+        relayHub: relayHub.address,
+        sender: relayHub,
+        owner,
+        factory: smartWalletFactory,
+        tokenContract: testToken.address,
+        isCustomSmartWallet: false,
+      });
```

Comment:
Nit: I'd move this call to the beginning of the test if possible, because I think this is a precondition for the test to be successful